src/components/CustomModal/index.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react';
+import { ComposedModal, ModalHeader, ModalBody, TextInput, Button, ModalFooter } from "carbon-components-react";
+import { postRequest, invItemURL } from "../../utils/api-utils";
+import styles from "./customModal.module.scss";
+
+const CustomModal = (props) => {
+    
+    const {
+        data,
+        showModal,
+        rootClass,
+        modalListClass,
+        subTitle,
+        primaryButton,
+        secondaryButton,
+        tabs,
+        handleSubmit,
+        closeModal } = props;
+    const [ modifiedData, setModifiedData ] = useState(data);
+    const [ postError, setPostError ] = useState('');
+    const [ editMode, setEditMode ] = useState({ "id": -1, "value": 0 });
+
+    /**
+     * description. send the post request with updated dispense as payload
+     * @namespace submitDespense
+     * @memberof  CustomModal
+     * @property {array} payload modified dispense data.
+    */
+
+    const submitDespense = async (payload) => {
+        const res = await postRequest(invItemURL, payload)
+        const responseData = await res.json();
+        if(res.error) {
+            setPostError(res.error.message);
+          } else {
+            handleSubmit(responseData);
+            closeModal(false)
+          }
+    }
+
+    /**
+     * description. handle the editable input field and update the dispense data obj
+     * @namespace handleEditMode
+     * @memberof  CustomModal
+     * @property {object} e current click event.
+     * @property {number} id index of modified dispense data.
+    */
+
+    const handleEditMode = (e, id) => {
+        e.preventDefault();
+        const targetValue = e && e.target && e.target.value;
+        if (targetValue && !/^[1-9]\d*(\.\d+)?$/.test(targetValue)) {
```

Comment:
Plz put regEx in a const, so its clear what the regEx is checking for. This is not obvious to me. : )

Reply:
will be taken care as part of https://trello.com/c/kLG4SL2k/28-inventory-dispensation-system-dispense-to-patients